README.md

```diff
@@ -39,17 +38,149 @@ julia> central_fdm(5, 1)(sin, 1) - cos(1)
+  samples:          10000
+  evals/sample:     195
+```
+
 Compute the second derivative of `sin` with a 5th order central method:
 
 ```julia
 julia> central_fdm(5, 2)(sin, 1) + sin(1)
--4.367495254342657e-11
+-8.767431225464861e-11
+```
+
+To obtain better accuracy, you can increase the order of the method:
+
+```julia
+julia> central_fdm(12, 2)(sin, 1) + sin(1)
+5.240252676230739e-14
 ```
 
 The functions `forward_fdm` and `backward_fdm` can be used to construct
 forward differences and backward differences respectively.
 
-Alternatively, you can construct a finite difference method on a custom grid:
+### Dealing with Singularities
+
+The function `log(x)` is only defined for `x > 0`.
+If we try to use `central_fdm` to estimate the derivative of `log` near `x = 0`,
+then we run into `DomainError`s, because `central_fdm` happens to evaluate `log`
+at some `x < 0`.
+
+```julia
+julia> central_fdm(5, 1)(log, 1e-3)
+ERROR: DomainError with -0.02069596546590111
+```
+
+To deal with this situation, you have two options.
+
+The first option is to use `forward_fdm`, which only evaluates `log` at inputs
+greater or equal to `x`:
+
+```julia
+julia> forward_fdm(5, 1)(log, 1e-3) - 1000
+-3.741856744454708e-7
+```
+
+This works fine, but the downside is that you're restricted to one-sided
+methods (`forward_fdm`), which tend to perform worse than central methods
+(`central_fdm`).
+
+The second option is to limit the distance that the finite difference method is
+allowed to evaluate `log` away from `x`. Since `x = 1e-3`, a reasonable value
+for this limit is `9e-4`:
+
+```julia
+julia> central_fdm(5, 1, max_range=9e-4)(log, 1e-3) - 1000
+-4.027924660476856e-10
+```
+
+Another commonly encountered example is `logdet`, which is only defined
+for positive-definite matrices.
+Here you can use a forward method in combination with a positive-definite
+deviation from `x`:
+
+```julia
+julia> x = diagm([1.0, 2.0, 3.0]); v = Matrix(1.0I, 3, 3);
+
+julia> forward_fdm(5, 1)(ε -> logdet(x .+ ε .* v), 0) - sum(1 ./ diag(x))
+-4.222400207254395e-12
+```
+
+A range-limited central method is also possible:
+
+```julia
+julia> central_fdm(5, 1, max_range=9e-1)(ε -> logdet(x .+ ε .* v), 0) - sum(1 ./ diag(x))
+-1.283417816466681e-13
+```
+
+### Richardson Extrapolation
+
+The finite difference methods defined in this package can be extrapolated using
+[Richardson extrapolation](https://github.com/JuliaMath/Richardson.jl).
+This can offer superior numerical accuracy:
+Richardson extrapolation attempts polynomial extrapolation of the finite
+difference estimate as a function of the step size until a convergence criterion
+is reached.
+
+```julia
+julia> extrapolate_fdm(central_fdm(2, 1), sin, 1)[1] - cos(1)
+1.6653345369377348e-14
+```
+
+Similarly, you can estimate higher order derivatives:
+
+```julia
+julia> extrapolate_fdm(central_fdm(5, 4), sin, 1)[1] - sin(1)
+-1.626274487942503e-5
+```
+
+In this case, the accuracy can be improved by lowering (making closer to `1`)
```

Comment:
You mean closer to ,0?

Reply:
No, I mean closer to `1`. I meant lowered as in making less aggressive. I'll change the wording.